### PR TITLE
feat: ZkSync Era

### PR DESCRIPTION
**Pull request by Olyno**
# Description

Hi 👋🏻 

I really wanted to implement the ZkSync Era network, which is currently missing. Not knowing exactly the whole process, I contributed based on your commit for PulseChain (https://github.com/tangem/blockchain-sdk-kotlin/commit/e1cab9395feb1baf1c407370dba741c451251486).

Feel free to tell me what's wrong, or even redirect me to parts I might have overlooked.

### CODE REVIEW

**Comment by iiiburnyiii** [comment]
Hi, Olyno!
I'm Anton, Android developer at Tangem. You did everything right, but we need to get confirmation from our team. I'll discuss it and get back to you. 
Thank you for your contribution!

**Comment by Sateetas** [comment]
Hey Olyno!

I'm Ilya, an Android developer at Tangem

First off, I want to thank you for creating this contribution. Everything looks excellent, however, there are some conflicting files. Could you kindly take a moment to resolve them?

Your contributions are very timely, as we are about to implement several new blockchains, with zkSync Era being one of the key ones we're implementing

**Comment by Olyno** [comment]
Hi Ilya,

I just resolved all conflicts. Don't hesitate to give me any additional feedback if needed.

Thank you!

File: blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt [hunk 1]
```diff
@@ -462,6 +468,7 @@ enum class Blockchain(
             AreonTestnet -> Chain.AreonTestnet.id
             PulseChain -> Chain.PulseChain.id
             PulseChainTestnet -> Chain.PulseChainTestnet.id
+            ZkSyncEraTestnet, Chain.ZkSyncEraTestnet.id,
```
Comment (Sateetas) [inline blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt]: Could you please replace first comma with `->` and remove last comma?

Comment (Olyno) [reply blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt]: Nevermind, fixed!

File: blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt [hunk 2]
```diff
@@ -124,6 +124,8 @@ enum class Blockchain(
     AreonTestnet("areon/test", "TAREA", "Areon Network Testnet"),
     PulseChain("pls", "PLS", "PulseChain"),
     PulseChainTestnet("pls/test", "tPLS", "PulseChain Testnet v4"),
+    ZkSyncEra("ZkSync-Era", "ETH", "ZkSync Era"),
```
Comment (Sateetas) [inline blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt]: And please make this ID start with a lowercase letter, according to this blockchain official naming

Comment (Sateetas) [reply blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt]: I think the best options are `zkSyncEra` for mainnet and `zkSyncEra/test` for testnet

Comment (Olyno) [reply blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt]: Done ✅

Comment (Sateetas) [reply blockchain/src/main/java/com/tangem/blockchain/common/Blockchain.kt]: Thanks!

**Review by Sateetas — APPROVED** [review]

**Review by kozarezvlad — APPROVED** [review]

**Review by MasterBin — APPROVED** [review]

**Pull request merged**